Added hamcrest and annotation

Uwampojejecharly opened the pull request:
Added hamcrest to do the TDD (Test Driven Developpement), created more powerful tests using annotations.

Uwampojejecharly committed:
Add new tests and discovered more about tests

In this commit i discovered more new topics as:
	Using some annotations like beforeClass,afterClass,...
	Using hamcrest matchers to do assertions.
	TDD

Uwampojejecharly committed:
Added hamcrest and annotation

Added hamcrest to do the TDD (Test Driven Developpement), created more
powerful tests using annotations.

fayabobo reviewed (changes requested):
Can you modify your commit message:

"Add new tests and discovered more about tests

In this commit I discovered more new topics as:
- Using some annotations like beforeClass, afterClass, ..
-  Using hamcrest matchers to do assertions.
-  TDD"

Uwampojejecharly committed:
Add new tests and discovered more about tests

In this commit I discovered more new topics as:

	Using some annotations like beforeClass, afterClass, ...
	Using hamcrest matchers to do assertions.
	TDD

fayabobo reviewed (changes requested):
- There is an issue with the JUnit and the hamcrest dependencies. Check the mail I sent you for more information.
- Remove the commons-lang3 library because it is not used here.

One important thing: When modifying commits, make sure you amend the old message using "git amend". You should not do a normal commit because you are modifying an existing one. After "git add ." use "git amend" to make sure you do it with the same message.

fayabobo reviewed (changes requested):
There is something wrong in the commit.

Can you modify it? See my old comment about the commit.
